### ISSUE TITLE
Consider using Task.Supervisor.async/4 for distributed tasks

### ISSUE THREAD

**Issue by fishcakez** [elixir-lang/elixir-lang.github.com#616]
Someone reported `BadFunctionError` problems on IRC, even though md5/compile time matched on both nodes.

**Comment by dabaer** [comment]
Using the example router from the docs I get the following error
https://gist.github.com/dabaer/209b4352c5f838f05c34

However, following @fishcakez suggestion, replacing Task.Supervisor.async/2 with Task.Supervisor.async/4 fixes the issue:

```elixir
Task.Supervisor.async(sup, KV.Router, :route, [bucket, mod, fun, args])
```

**Comment by josevalim** [comment]
Fixed in upcoming 1.2 branch.
